src/components/workspace/editorMode/source.js

```diff
@@ -63,63 +63,99 @@ function _DocCommentHighlightRules(acequire, exports, module) {
 function _SourceHighlightRules(acequire, exports, module) {
     "use strict";
     
+    //parent classes to be inherited
     const oop = acequire("../lib/oop");
     const DocCommentHighlightRules = acequire("./doc_comment_highlight_rules").DocCommentHighlightRules;
     const TextHighlightRules = acequire("./text_highlight_rules").TextHighlightRules;
-    const identifierRegex = "[a-zA-Z\\$_\u00a1-\uffff]-[a-zA-Z\\d\\$_\u00a1-\uffff]*";
-
+    
```

Comment:
Avoid repetition of names. The Source languages form a strict hierarchy, so a predeclared name in Source §x is also a predeclared name in Source §(x+i), i > 0. 

You should exploit this by concatenating the names:
const chapter2 = chapter1 + "...";
etc

Reply:
i see. i will make changes to it